jinja2/filters.py

```diff
@@ -457,7 +457,12 @@ def do_indent(s, width=4, indentfirst=False):
             indent by two spaces and indent the first line too.
     """
     indention = u' ' * width
-    rv = (u'\n' + indention).join(s.splitlines())
+    if indentblanklines:
+        rv = (u'\n' + indention).join(s.splitlines())
+    else:
+        lines = s.splitlines()
+        rv = lines.pop(0) + u'\n'
```

Comment:
This adds a trailing newline if there is only one line, unlike `indentblanklines=True`.